### PR TITLE
Update client to 10.90 version

### PR DESCRIPTION
**Pull request by Tulioh**

### CODE REVIEW

File: src/protocolgame.cpp [hunk 1]
```diff
@@ -2388,18 +2398,17 @@ void ProtocolGame::sendAddCreature(const Creature* creature, const Position& pos
 	}
 
 	msg.addByte(0x00); // can change pvp framing option
-	msg.addByte(0x00); // expert mode button enabled
+	msg.addByte(g_game.getWorldType() != WORLD_TYPE_PVP_ENFORCED ? 0x01 : 0x00);
```
Comment (WibbenZ) [inline src/protocolgame.cpp]: Since the no one has spent time to actually write new new war system, there is no reason for us to even use the pvp enfo check.
Just send 0x00 and be happy with it xD
The buttons won't do anything anyways, since there is no parser function for them (kinda~).

File: src/protocolgame.cpp [hunk 2]
```diff
@@ -2388,18 +2398,17 @@ void ProtocolGame::sendAddCreature(const Creature* creature, const Position& pos
...
 
 	sendPendingStateEntered();
 	sendEnterWorld();
 	sendMapDescription(pos);
-
-	if (isLogin) {
-		sendMagicEffect(pos, CONST_ME_TELEPORT);
-	}
-
+
```
Comment (WibbenZ) [inline src/protocolgame.cpp]: Trailing

File: src/protocolgame.cpp [hunk 3]
```diff
@@ -2388,18 +2398,17 @@ void ProtocolGame::sendAddCreature(const Creature* creature, const Position& pos
...
+	msg.addString("http://static.tibia.com/images/store/");
+	msg.addByte(g_config.getNumber(ConfigManager::MAX_PACKETS_PER_SECOND));
 
 	writeToOutputBuffer(msg);
 
 	sendPendingStateEntered();
 	sendEnterWorld();
 	sendMapDescription(pos);
-
-	if (isLogin) {
```
Comment (WibbenZ) [inline src/protocolgame.cpp]: Why?

File: src/protocolgame.cpp [hunk 4]
```diff
@@ -2351,22 +2362,25 @@ void ProtocolGame::sendAddCreature(const Creature* creature, const Position& pos
...
+		bool known;
+		uint32_t removedKnown;
+		checkCreatureAsKnown(creature->getID(), known, removedKnown);
+		AddCreature(msg, creature, known, removedKnown);
+		writeToOutputBuffer(msg);
+
 		if (isLogin) {
 			sendMagicEffect(pos, CONST_ME_TELEPORT);
 		}
+
```
Comment (ranisalt) [inline src/protocolgame.cpp]: why?

File: src/protocolgame.cpp [hunk 5]
```diff
@@ -2399,7 +2416,7 @@ void ProtocolGame::sendAddCreature(const Creature* creature, const Position& pos
 	if (isLogin) {
 		sendMagicEffect(pos, CONST_ME_TELEPORT);
 	}
-
+
```
Comment (ranisalt) [inline src/protocolgame.cpp]: trailing whitespace

(@wibbenz sorry bro you're late)

File: src/protocolgame.cpp [hunk 6]
```diff
@@ -2390,6 +2404,9 @@ void ProtocolGame::sendAddCreature(const Creature* creature, const Position& pos
 	msg.addByte(0x00); // can change pvp framing option
 	msg.addByte(0x00); // expert mode button enabled
 
+	msg.addString("http://static.tibia.com/images/store/");
+	msg.addByte(g_config.getNumber(ConfigManager::MAX_PACKETS_PER_SECOND));
```
Comment (Luckey1729) [inline src/protocolgame.cpp]: ``MAX_PACKETS_PER_SECOND``

So.. 25?

How is this value related to the 10.90 update? Did something change with sendAddCreature and MAX_PACKETS_PER_SECOND?

Comment (Tulioh) [inline src/protocolgame.cpp]: Yeah, these lines are needed.

Comment (TheSumm) [inline src/protocolgame.cpp]: Using the value of MAX_PACKETS_PER_SECOND is wrong though.

Comment (Tulioh) [inline src/protocolgame.cpp]: What is you suggestion?

2016-02-13 14:24 GMT-02:00 Sam <notifications@github.com>:

Comment (TheSumm) [inline src/protocolgame.cpp]: create a new config value coinPacketSize = 25 and use that one.

File: src/protocolgame.cpp [hunk 7]
```diff
@@ -2351,22 +2362,25 @@ void ProtocolGame::sendAddCreature(const Creature* creature, const Position& pos
...
+
+		bool known;
+		uint32_t removedKnown;
+		checkCreatureAsKnown(creature->getID(), known, removedKnown);
+		AddCreature(msg, creature, known, removedKnown);
+		writeToOutputBuffer(msg);
+
 		if (isLogin) {
-			sendMagicEffect(pos, CONST_ME_TELEPORT);
+		    sendMagicEffect(pos, CONST_ME_TELEPORT);
```
Comment (djarek) [inline src/protocolgame.cpp]: Use tabs instead of spaces for indents.

Comment (ranisalt) [inline src/protocolgame.cpp]: There's an .editorconfig file in the repository exactly because of that. Please enable support on your editor if available.

File: src/protocolgame.cpp [hunk 8]
```diff
@@ -2390,6 +2404,9 @@ void ProtocolGame::sendAddCreature(const Creature* creature, const Position& pos
 	msg.addByte(0x00); // can change pvp framing option
 	msg.addByte(0x00); // expert mode button enabled
 
+	msg.addString("http://static.tibia.com/images/store/");
```
Comment (esfomeado) [inline src/protocolgame.cpp]: I think that you should removed this reference from Tibia.

Comment (vankk) [inline src/protocolgame.cpp]: why?

Comment (ranisalt) [inline src/protocolgame.cpp]: Copyright

File: src/protocolgame.cpp (hunk 6 above)
Comment (WibbenZ) [inline src/protocolgame.cpp]: Im not fully gonna take credit for these fails, but when I started to compare the changes and noticed 25 and rememberd the max package config value.
And I wrote abt it in the 1090 thread....
But anyways do as summ said and create a new config integer.

File: src/protocolgame.cpp (hunk 8 above)
Comment (WibbenZ) [inline src/protocolgame.cpp]: You also make it harder for ppl wanting to replace the website store.
Use a config value here aswell, default it to
.... .com/store/
Or smth like that

File: src/protocolgame.cpp (hunk 6 above)
Comment (Tulioh) [inline src/protocolgame.cpp]: This config integer need to be editable on config.lua?

File: src/configmanager.h [hunk 9]
```diff
@@ -103,6 +104,7 @@ class ConfigManager
 			MAX_MARKET_OFFERS_AT_A_TIME_PER_PLAYER,
 			EXP_FROM_PLAYERS_LEVEL_RANGE,
 			MAX_PACKETS_PER_SECOND,
+			COIN_PACKET_SIZE,
```
Comment (WibbenZ) [inline src/configmanager.h]: You are missing this in your luascript.cpp file.
Please test the things you upload.

**Comment by WibbenZ** [comment]
New mounts and outfits are also missing, if you need help with them send me a PM on otland and ill send you the ids.

File: config.lua [hunk 10]
```diff
@@ -70,6 +70,7 @@ kickIdlePlayerAfterMinutes = 15
 maxMessageBuffer = 4
 emoteSpells = false
 classicEquipmentSlots = false
+coinImagesURL = "http://some.url/images/store"
```
Comment (WibbenZ) [inline config.lua]: Missing coinPacketSize

**Comment by slavidodo** [comment]
If you are updating to 10.90, add the already missing features in 10.82 instead of just updating to a new client without any purposes.

I don't disagree with the update but this shouldn't be a pull request. As a pull request including a client update should have the new client features. 

BTW, I 've already tried to make the global depot but it seems that tibia changed the depot chest itself not just adding depots inside it!

**Comment by slavidodo** [comment]
Btw, I am going to develop global depot, just i need information about it, does tibia removed city's depot id? every depot can be opened from any city now?

File: config.lua (hunk 10 above)
Comment (Tulioh) [inline config.lua]: didn't coinPacketSize is a internal feature?

Comment (WibbenZ) [inline config.lua]: Not sure what you mean.
But you are supposed to be able to change it, 25 is the amount you can send in one package (you can't transfer 24 or 26 to a player, you can only transfer / buy / sell in an increment of 25).

In OTs we might wanna change this, ex be able to transfer with an increment of 1 or if we use really high prices 1000 etc etc
And since you also added it to the source files you should add it to the config.lua file, even if it will default to 25 you should still make it easier to newbies to change it.

And id also add them to a new line like the house configs and maybe explain what they do, that what the URL should be etc etc

**Comment by WibbenZ** [comment]
Haven't checked the updates with the depot, but in my understanding they removed all the depots in a way, you now have 1 depot (like the inbox, just copy the code) then add the new depots (idk the item ids) into that box, those boxes should get the town ids.

Our code probbly still works, since they most likely just changed the sprite of the depot box (next to the inbox) and then added new sprites for each town id.

**Comment by vankk** [comment]
> does tibia removed city's depot id? every depot can be opened from any city now?

Yes.

-     Thais - Town ID: 2
-     Carlin - Town ID: 4
-     Kazordoon - Town ID: 3
-     Ab'Dendriel - Town ID: 5
-     Edron - Town ID: 11
-     Darashia - Town ID: 10
-     Venore - Town ID: 1
-     Ankrahmun - Town ID: 9
-     Port Hope - Town ID: 8
-     Liberty Bay - Town ID: 7
-     Svargrond - Town ID: 12
-     Yalahar - Town ID: 13
-     Travora - Town ID:
-     Farmine - Town ID: 14
-     Gray Beach - Town ID: 28
-     Roshamuul - Town ID: 29
-     Rathleton - Town ID: 33

**Comment by Tulioh** [comment]
Not sure about this. Now has depots representing a city, but can be opened
in any city.

2016-02-19 11:40 GMT-02:00 vankk <notifications@github.com>:

**Comment by TheSumm** [comment]
Depots don't represent a city anymore. You just have a global depot with 17 boxes inside.

**Comment by Tulioh** [comment]
But is one depot for each city.

**Comment by ranisalt** [comment]
Well, it's should not be that hard. The function that returns the depot of a city could be refactored to return a (ordered) set with all depots instead.

**Comment by TheSumm** [comment]
The new boxes are not bound to cities in any way. If they add new cities that does not mean more boxes are added (see Krailos).

**Comment by WibbenZ** [comment]
Well in a way they are bound to a city ex. 1 = thais, but a city does not require a depot chest if that is what you ment.
But this should be really simple, change the depot chest into an inbox chest = you can't add any items etc
Then in that box add 17 more boxes with the correct sprites that are the depots - mission completed :p

**Comment by slavidodo** [comment]
Okay, I have started on that.

Can someone please try to update the remere's map editor ( removing depot id  for 10.80 +)

File: src/protocolgame.cpp [hunk 11]
```diff
@@ -2351,22 +2362,25 @@ void ProtocolGame::sendAddCreature(const Creature* creature, const Position& pos
...
-			msg.addByte(0x6A);
-			msg.addPosition(pos);
-			msg.addByte(stackpos);
-
-			bool known;
-			uint32_t removedKnown;
-			checkCreatureAsKnown(creature->getID(), known, removedKnown);
-			AddCreature(msg, creature, known, removedKnown);
-			writeToOutputBuffer(msg);
+		if (stackpos >= 10) {
```
Comment (slavidodo) [inline src/protocolgame.cpp]: I think this will make some problems with @djarek Casting System.

Comment (vankk) [inline src/protocolgame.cpp]: That is really sad to read

**Comment by slavidodo** [comment]
@marksamman Should pagination work for any container (e.g depot) or just developed for inbox?

**Comment by slavidodo** [comment]
2 problems I faced: 
- Pagination isn't working as expected.
- When you stand on depot tile, it only tells the number of items in depot (1) : `Your depot contains x items.`

**Comment by Kamenuvol** [comment]
If someone updates his server, it should put items from depotID 1 to depot box 1, and so on.
If there are more than 17 towns in his server the rest should go to inbox.

**Comment by comedinha** [comment]
Some fixes and no-oficial items.otb https://github.com/Tulioh/forgottenserver/pull/1

**Comment by comedinha** [comment]
Pagination is working here...
http://i.imgur.com/cPYxjJ8.jpg?1
http://i.imgur.com/LookbRU.jpg?1

**Comment by WibbenZ** [comment]
Well ex depots have pagination now aswell.

**Comment by slavidodo** [comment]
I have already tried too much, pagination didn't work within depots while
it's normally working with inbox, browse field, ..etc.
And that's the reason why I stopped, if someone wanna continue from where i
stopped, no problem.

-- About the Premium shop.

I already made it but still misses some small things like store errors,
transferring coins.

**Comment by comedinha** [comment]
Initial new depot system
Pagination have 1 little problem - When you chanche the page not is possible back to depot chest (is protocolgame problem? - only need open the chest again)
If the server use depotid 0 he go to town 17, if you server have > 17 towns, all items go to town 17.

https://github.com/Tulioh/forgottenserver/pull/2

**Comment by comedinha** [comment]
@marksamman Allow the official items.otb to start create the xml and others ;-;

**Comment by slavidodo** [comment]
Seems your pagination worked, what is still the problem?

**Comment by comedinha** [comment]
@slavidodo Fixed pagination and change the hasPagination

**Comment by Tazer89** [comment]
@comedinha its your otb official items otb that tfs will use for sure?

**Comment by comedinha** [comment]
@Tazer89 no, I'm hoping that the @marksamman launch an oficial items.otb to start work on items...

**Comment by Tazer89** [comment]
Ok thanks for fast answer. :)

**Comment by comedinha** [comment]
@Tazer89 No have problems on create server, only not use new items... (outfits, mounts and others is ok)

**Comment by Tazer89** [comment]
Hey guys, i just wonder if there are any fix for the premium days? Becouse when you are premium you have alot of days even if you just has 3.

**Comment by gesior** [comment]
What about 'equip' hotkey? Anyone analysed Tibia 10.81+ protocol and added it?

File: src/itemloader.h [hunk 12]
```diff
@@ -102,6 +102,7 @@ enum clientVersion_t {
 	CLIENT_VERSION_1031 = 54,
 	CLIENT_VERSION_1035 = 55,
 	CLIENT_VERSION_1076 = 56,
+	CLIENT_VERSION_1091 = 57,
```
Comment (slavidodo) [inline src/itemloader.h]: You are missing this,
```c++
CLIENT_VERSION_1080 = 57,
CLIENT_VERSION_1082 = 58,
CLIENT_VERSION_1090 = 59,
CLIENT_VERSION_1091 = 60
```

Comment (comedinha) [inline src/itemloader.h]: 80, 82, 90 and 91 not are oficial, the next oficial number is 57...

**Comment by comedinha** [comment]
Have a little problem with new depots and market... I need rework the depot code...

**Comment by slavidodo** [comment]
Edit this function https://github.com/otland/forgottenserver/blob/master/src/protocolgame.cpp#L1563 with this code.
``` c++
NetworkMessage msg;
	msg.addByte(0xF6);

	msg.add<uint64_t>(player->getBankBalance());
	msg.addByte(std::min<uint32_t>(IOMarket::getPlayerOfferCount(player->getGUID()), std::numeric_limits<uint8_t>::max()));

	std::map<uint16_t, uint32_t> depotItems;
	std::forward_list<Container*> containerList { player->getInbox() };

	for (int16_t i = 0; i < 17; i++) {
		auto depotChest = player->getDepotChest(i, true);
		if (!depotChest) {
			continue;
		}
		containerList.push_front(depotChest);
	}

	player->setInMarket(true);
```

**Comment by slavidodo** [comment]
@comedinha Have you tested leaving house also?

File: src/protocolgame.cpp [hunk 13]
```diff
@@ -2390,6 +2403,9 @@ void ProtocolGame::sendAddCreature(const Creature* creature, const Position& pos
 	msg.addByte(0x00); // can change pvp framing option
 	msg.addByte(0x00); // expert mode button enabled
 
+	msg.addString(g_config.getString(ConfigManager::COIN_IMAGES_URL));
+	msg.addByte(g_config.getNumber(ConfigManager::COIN_PACKET_SIZE));
```
Comment (slavidodo) [inline src/protocolgame.cpp]: Actually this is u16 not byte.
```
msg.add<uint16_t>(static_cast<uint16_t>(g_config.getNumber(ConfigManager::COIN_PACKET_SIZE)));
```
